aten/src/ATen/mps/MPSDevice.h

```diff
@@ -48,11 +54,14 @@ class TORCH_API MPSDevice {
     return _mtl_device;
   }
 
+  MTLFunction_t metalIndexingFunction(const std::string &kernel, MTLFunctionConstantValues_t constantValues);
+
   ~MPSDevice();
 
  private:
   static MPSDevice* _device;
   MTLDevice_t _mtl_device;
+  MTLLibrary_t _mtl_indexing_library;
```

Comment:
Why does it need to be a part of MPSDevice rather than implementation detail in Indexing.mm?

Reply:
This is cached on the MPSDevice itself the very first time index.Tensor_out is called  - all the subsequent calls will use directly the cached version of the library

Reply:
No, just from code clarify point of view - mtl_indexing_library is implementation detail of Indexing operator and should not leak into MPSDevice. We can implement a RAII mechanism of registering libraries with MPSDevice, but individual implementations IMO do not belong here.